snorkel/utils.py

```diff
@@ -75,25 +83,29 @@ def matrix_coverage(L):
     Given an N x M matrix where L_{i,j} is the label given by the jth LF to the ith candidate:
     Return the **fraction of candidates that each LF labels.**
     """
-    return np.ravel(sparse_abs(L).sum(axis=0) / float(L.shape[0]))
+    return np.ravel(sparse_nonzero(L).sum(axis=0) / float(L.shape[0]))
 
 
 def matrix_overlaps(L):
     """
     Given an N x M matrix where L_{i,j} is the label given by the jth LF to the ith candidate:
     Return the **fraction of candidates that each LF _overlaps with other LFs on_.**
     """
-    L_abs = sparse_abs(L)
-    return np.ravel(np.where(L_abs.sum(axis=1) > 1, 1, 0).T * L_abs / float(L.shape[0]))
+    L_nonzero = sparse_nonzero(L)
+    return np.ravel(np.where(L_nonzero.sum(axis=1) > 1, 1, 0).T * L_nonzero / float(L.shape[0]))
 
 
 def matrix_conflicts(L):
     """
     Given an N x M matrix where L_{i,j} is the label given by the jth LF to the ith candidate:
     Return the **fraction of candidates that each LF _conflicts with other LFs on_.**
     """
-    L_abs = sparse_abs(L)
-    return np.ravel(np.where(L_abs.sum(axis=1) != sparse_abs(L.sum(axis=1)), 1, 0).T * L_abs / float(L.shape[0]))
+    B = L.copy()
+    for row in range(0,B.shape[0]):
+        if np.unique(B.getrow(row).data).size == 1:
+            B.data[B.indptr[row]:B.indptr[row+1]] = 0
+    return matrix_coverage(sparse_nonzero(B))
+
```

Comment:
Is this slow because of the looping over rows of B?  Probably should be fine... can come back to this I guess, happy to test later